packages/commonwealth/client/scripts/views/pages/view_proposal/index.tsx

```diff
@@ -64,8 +65,32 @@ const ViewProposalPage = ({
     };
   }, [proposal, forceRerender]);
 
+  const onAave = app.chain?.network === ChainNetwork.Aave;
+  const fetchAaveData = onAave && isAdapterLoaded;
+  const { data: cachedAaveProposals, isLoading: aaveProposalsLoading } =
+    useAaveProposalsQuery({
+      moduleReady: fetchAaveData,
+      chainId: app.chain?.id,
+    });
+
+  if (aaveProposalsLoading === false && fetchAaveData && !proposal) {
+    const foundProposal = cachedAaveProposals?.find(
+      (p) => p.identifier === proposalId
+    );
+
+    if (!foundProposal?.ipfsData) {
+      foundProposal.ipfsDataReady.on('ready', () => {
```

Comment:
this is not a proper way of setting up a listener.  This code gets executed on every rerender so it means that you will setup listener multiple times (on every rerender) which leads to memory leak problem. Also, you are missing the removal of event listener. Usually this kind of logic sits in useEffect

Reply:
I moved this logic into `useEffect` and switched to `.once`